src/pages.c

```diff
@@ -193,7 +193,10 @@ pages_map(void *addr, size_t size, size_t alignment, bool *commit) {
 		if (addr != NULL) {
 			flags |= MAP_FIXED | MAP_EXCL;
 		} else {
-			flags |= MAP_ALIGNED(ffs(alignment) - 1);
+			int alignment_bits = ffs(alignment);
```

Comment:
Ditto with `ffs_zu`.

Probably even better, we should assert the alignment; we should always have tweaked it as necessary before we get to the pages module.